packages/swingset-liveslots/src/virtualObjectManager.js

```diff
@@ -1005,46 +1026,33 @@ export const makeVirtualObjectManager = (
 
       proto = defendPrototypeKit(
         tag,
-        // TODO Once agoric-sdk is upgraded to depend on an `@endo/exo`
-        // incorporating https://github.com/endojs/endo/pull/1966
-        // Then the following at-ts-ignore will no longer be needed.
-        // However, it is an at-ts-ignore rather than an
-        // at-ts-expect-error to be compat with endo both before and
-        // after, until we're safely across the transition.
-        //
-        // eslint-disable-next-line @typescript-eslint/prefer-ts-expect-error
-        // @ts-ignore
-        harden(contextProviderKit),
+        harden(contextProviderKitVar),
         behavior,
         thisfulMethods,
         interfaceGuardKit,
       );
     } else {
-      /** @type {ClassContextProvider} */
-      const contextProvider = rep => {
+      contextProviderVar = rep => {
         const slot = getSlotForVal(rep);
-        assert(slot !== undefined);
+        if (slot === undefined) {
+          return undefined;
+        }
```

Comment:
@FUDCo , this is a semantic change to existing virtual kind functionality. Is it still correct?

Reply:
I checked, and only my own code consumes this, which I know is prepared for these `undefined`s. So it cannot be an incorrect change based on code I don't know. Resolving.